src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml

```diff
@@ -114,4 +114,17 @@
           p:workflowDocumentActionsService-ref="workflowDocumentActionsService"
     />
 
+    <!--
+        Override processing queue service to fix an issue that can cause the workflow engine
+        to process a stale copy of the route header and/or its downstream data.
+     -->
+    <bean id="documentProcessingQueue" class="org.kuali.kfs.kew.impl.document.DocumentProcessingQueueImpl"
+          p:workflowEngineFactory-ref="workflowEngineFactory"
+          p:actionRequestService-ref="actionRequestService"
+          p:documentAttributeIndexingQueue-ref="documentAttributeIndexingQueue"
+          p:routeHeaderService-ref="documentRouteHeaderService"
+          p:routeNodeService-ref="routeNodeService"
+          p:ksbMessageDeliveryMode="${message.delivery}"
```

Comment:
When I look for message.delivery in kuali-puppet, I only see it in some bash SH files.  Is this setting needed elsewhere

Reply:
In the financials project, the default KFS config file sets `message.delivery` to asynchronous mode, whereas the default integration test config file sets it to synchronous mode. These defaults should be appropriate and we don't need to override them elsewhere.

If you notice a spot where a non-unit-test environment is trying to set up synchronous processing instead, please let me know.

Reply:
Take a look at https://github.com/search?q=repo%3ACU-CommunityApps%2Fkuali-puppet%20message.delivery&type=code .... looks like batch_job_up.sh is used, we should double check that

Reply:
looking at the code in https://github.com/CU-CommunityApps/kuali-puppet/blob/5bbd11f427df3d607ad9fd2f421c5de046edc016/files/batch_scripts/batch_job_up.sh#L22 I don't think it would ever set  mode="-Dmessage.delivery=synchronous"

Reply:
I agree that I'm not seeing any setup that would result in the `-Dmessage.delivery=synchronous` option getting chosen, unless I missed something. Also, I confirmed in the KFS DEV logs that this morning's eInvoice Route Job was performing the `StandardWorkflowEngine` work in a different thread than the one that routed the document. It seems like we already have asynchronous processing set up in the KFS batch system.